lib/SILOptimizer/Transforms/AccessEnforcementOpts.cpp

```diff
@@ -398,16 +398,19 @@ void AccessConflictAndMergeAnalysis::addOutOfScopeAccess(
     // We don't have a match in outOfScopeConflictFreeAccesses
     // Just add it and return
     info.outOfScopeConflictFreeAccesses.conflictFreeAccesses.insert(
-        std::make_pair(newIndex, beginAccess));
+        beginAccess);
     return;
   }
 
-  auto *otherBegin = it->second;
-  auto index = it->first;
-  info.outOfScopeConflictFreeAccesses.conflictFreeAccesses.erase(index);
+  auto *otherBegin = *it;
+  auto rmIt = std::find(
```

Comment:
Why is this now using `find` instead of `conflictFreeAccesses.erase(otherBegin)`?

Reply:
You can't use `erase` on a non-iterator type. `it` is a reverse iterator.